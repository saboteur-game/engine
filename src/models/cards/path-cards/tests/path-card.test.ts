import Position from "../../../position";
import CardParameters from "../../card-parameters";
import PathCard from "../path-card";

const PLAY_POSITION = new Position(0, 1);

describe("PathCard", () => {
  let pathCard: PathCard;
  let cardParameters: CardParameters;

  beforeEach(() => {
    pathCard = new PathCard([1, 2, 3, 4]);
    cardParameters = { position: PLAY_POSITION };
  });

  it("can be serialized", () => {
    expect(pathCard.toJS()).toMatchSnapshot();
  });

  it("can be rotated", () => {
    expect(pathCard.isUpsideDown).toBe(false);
    pathCard.rotate();
    expect(pathCard.isUpsideDown).toBe(true);
  });

  describe("play a card", () => {
    it("returns the played card", () => {
      expect(pathCard.play(cardParameters)).toBe(pathCard);
    });
  });
});
