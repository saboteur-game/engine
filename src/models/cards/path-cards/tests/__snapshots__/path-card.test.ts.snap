// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PathCard can be serialized 1`] = `
Object {
  "connectors": Array [
    1,
    2,
    3,
    4,
  ],
  "id": "test-id-1",
  "isUpsideDown": false,
  "parameters": undefined,
  "status": "unused",
}
`;
